lib/geordi/gitlinear.rb

```diff
@@ -0,0 +1,201 @@
+require 'highline'
+require 'net/http'
+require 'json'
+require 'active_support/core_ext/object/blank'
+
+module Geordi
+  class Gitlinear
+    # This require-style is to prevent Ruby from loading files of a different
+    # version of Geordi.
+    require File.expand_path('settings', __dir__)
+
+    API_ENDPOINT = 'https://api.linear.app/graphql'.freeze
+
+    def initialize
+      self.highline = HighLine.new
+      self.settings = Settings.new
+    end
+
+    def run_commit(git_args)
+      Interaction.warn <<~WARNING unless Util.staged_changes?
+        No staged changes. Will create an empty commit.
+      WARNING
+
+      issue = choose_issue
+      if issue
+        create_commit "[#{issue['identifier']}] #{issue['title']}", "Issue: #{issue['url']}", *git_args
+      end
+    end
+
+    def run_branch(from_master: false)
+      issue = choose_issue || Interaction.fail('No issue selected.')
+
+      branch_list_string = if Util.testing?
+                             ENV['GEORDI_TESTING_GIT_BRANCHES'] || ''
+                           else
+                             `git branch --format="%(refname:short)"`
+                           end
+
+      if branch_list_string.nil? || branch_list_string.strip.empty?
+        Interaction.fail 'Could not determine local git branches.'
+      end
+
+      new_branch_name = "#{issue['branchName']}"
+
+      local_branches = branch_list_string.split("\n")
+
+      branch_name = local_branches.find { |branch_name| branch_name == new_branch_name }
+      branch_name ||= local_branches.find { |branch_name| branch_name.include? issue['identifier'].to_s }
+
+      if branch_name.present?
+        checkout_branch branch_name, new_branch: false
+      else
+        checkout_branch new_branch_name, new_branch: true, from_master: from_master
+      end
+    end
+
+    private
+
+    attr_accessor :highline, :settings
+
+    def choose_issue
+      loading_message = 'Connecting to Linear ...'
+      print(loading_message)
+      issues = applicable_issues
+      reset_loading_message = "\r#{' ' * (loading_message.length + issues.length)}\r"
+
+      Geordi::Interaction.fail('No issues to offer.') if issues.empty?
+
+      if Util.testing?
+        return issues[0]
+      end
```

Comment:
If tests exit here anyway, there's no much gain in having a test switch in `applicable_issues` as well. Instead, please move the `Util.testing?` to the top of this method and do all required stuff directly there:
- Invoke `settings.linear_api_key` (presumably to see if settings are present)
- Return what L116+ does
Also, should the test identifier not be "team-123" for Linear? Afaik they are not numbers any more.

Reply:
You are right, I renamed the identifier to "team-123"